### DISCUSSION TITLE
M_uuid as a candidate for the fpm registry?

### DISCUSSION THREAD

**Discussion by urbanjost** [fortran-lang/fpm#582] [General]
I have a module that generates UUIDs that I use quite a bit that has an open license but I did not originate, albeit I added a CI/CD unit test and quite a bit of documentation.  There are the original sources and several other github repositories that contain the same basic code.

I also changed the code, but mostly some modest changes for portability.

So the question is not really if this is suitable, that can be determined by the PR request. The question is whether something licensed like this and with a similar pedigree is a suitable candidate.

I would feel no issues with something like BLAS that is public domain, but have not submitted this one because I did not originate it. On the other hand it is generally useful and I think I added some desirable packaging (fpm, of course) and documentation and testing features that make it suitable for submitting to the registry.  What do others think?

For completeness, the [github site](https://github.com/urbanjost/M_uuid)

I have other more substantial packages not as complete as this one that I am wondering whether I want to spend time packaging up.

**Comment by awvwgk** [comment]
This would be licensed under [BSD-4-Clause](https://spdx.org/licenses/BSD-4-Clause.html) by the original authors. It is a free permissive license. You can of course add your changes as MIT on top, which makes the total license something like `"BSD-4-Clause AND MIT"` or you just license your changes similarly as well.

**Reply by urbanjost** [reply]
A perfect match, even though withdrawn in 1999!  At least I can label it properly. I will just mark the rest the same. Thanks.

I keep meaning to write one from scratch but this one works fine, and that is what I hope some people are doing with some things I put out in the wild, so I guess I have an overdose of "not invented here" on this one.  I have a dozen or so similar I think I will try to bundle up for github/fpm in a form suitable for candidates on the fpm registry.  I am going to campaign some of the original authors to host it first, I think.
